### ISSUE TITLE
Update development/policy test-standard docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2748]
|<img alt="gpaci's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| gpaci reported|
|-|-|
|Trac ID|trac#2748|
|Type|defect|
|Created|2007-07-22 15:58:51Z|
|Branch|https://github.com/twisted/twisted/tree/update-test-std-2748|

doc/development/policy/test-standard.xhtml is out of date:

1) "Tests go in Twisted/twisted/test/" is no longer true (at least of lore, conch, etc.)

2) Links should be included to all the test directories, not just twisted/test

3) No mention is made of the Ultimate Quality Development System

Someone who understands what test_ files should go where can quickly fix (1) & (2);
mentioning UQDS near the beginning and adding a link to its definitive description (the wiki page?) would fix (3), though an outline of UQDS would probably be helpful, too.

4) "espouses on" should be "expounds on"

5) Add links to http://twistedmatrix.com/trac/wiki/TwistedTrial, http://twistedmatrix.com/documents/current/api/twisted.trial.html , doc/core/man/trial.1 , and any other Trial docs.

6) Mention that Trial can give line coverage information, which is very useful to ensure old code has decent coverage.

<details><summary>Searchable metadata</summary>

```
trac-id__2748 2748
type__defect defect
reporter__gpaci gpaci
priority__normal normal
milestone__ 
branch__branches_update_test_std_2748 branches/update-test-std-2748
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__documentation documentation
time__1185119931000000 1185119931000000
changetime__1226845561000000 1226845561000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [25125]) Branching to 'update-test-std-2748'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Fixed some broken links in r25126 and addressed the points above in r25127.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

The sentence about test coverage is inserted at a very awkward point.  Please move it to after the section about the todo feature, give it a `<h3>` of its own (like skip and todo have) and expand it a bit (perhaps talk about the option to pass, what the results are, etc).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [exarkun](#note_4):
> The sentence about test coverage is inserted at a very awkward point.  Please move it to after the section about the todo feature, give it a `<h3>` of its own (like skip and todo have) and expand it a bit (perhaps talk about the option to pass, what the results are, etc).

Thanks for the review, wasn't sure where to put that one so added it quite randomly, sorry about that.

Moved it into a section in r25318 and added the information that ```trial --help``` shows. Any other details should be added by someone with more insight, if necessary.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @thijstriemstra|
|-|-|

Cool, just one nit:

```
+    <code>Twisted/twisted/test/</code> or <code>Twisted/twisted/conch/test/</code>
```

Mentioning `Twisted` here is slightly misleading, twisted/test and twisted/conch/test is enough.

Please merge once done.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [therve](#note_6):
> Mentioning `Twisted` here is slightly misleading, twisted/test and twisted/conch/test is enough.

Fixed in r25372

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [25373]) Merge update-test-std-2748: General updates for the test-standard document.

Author: thijs
Reviewer: exarkun, therve
Fixes: [#2748](https://github.com/twisted/twisted/issues/2748)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
